filebeat/crawler/crawler.go

```diff
@@ -21,6 +21,8 @@ type Crawler struct {
 	prospectorConfigs   []*common.Config
 	out                 channel.Factory
 	wg                  sync.WaitGroup
+	ProspectorsFactory  cfgfile.RunnerFactory
```

Comment:
Note for myself: Need to understand why we need to store the factory now in the Crawler struct.

Reply:
Short answer: we use this factory from Autodiscovery. In the long term, we should unify all prospector/module initialization in a common place